browser/ntp_background/view_counter_service_factory.cc

```diff
@@ -79,6 +80,9 @@ ViewCounterServiceFactory::BuildServiceInstanceForBrowserContext(
         browser_context, std::make_unique<NTPBackgroundImagesSource>(service));
     content::URLDataSource::Add(
         browser_context, std::make_unique<NTPSponsoredImagesSource>(service));
+    content::URLDataSource::Add(
+        browser_context,
+        std::make_unique<NTPSponsoredRichMediaSource>(service->AsWeakPtr()));
```

Comment:
why is this weakptr? The service will outlive any webui

Reply:
Good catch, removed. Thanks

Reply:
also why are we adding these in the ViewCounterServiceFactory? Shouldn't these be in the webui? This is definitely not the right place for this

Reply:
This is based on legacy code as you can see, happy to raise a followup, thanks

Reply:
As discussed, raised https://github.com/brave/brave-browser/issues/43828. Thanks